wlauto/utils/types.py

```diff
@@ -300,3 +300,12 @@ def extend(self, values):
     def __str__(self):
         return ' '.join(self)
 
+
+class range_dict(dict):
+
+    def __getitem__(self, i):
+        i = int(i)
+        while i not in self and i > 0:
+            i -= 1
+        if i > 0:
+            return dict.__getitem__(self, i)
```

Comment:
I would also implement a `__setitem__` to make sure that only integers can be used as keys.